drivers/rtc/rtc_ifx_cat1.c

```diff
@@ -0,0 +1,349 @@
+ */
+
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <cy_pdl.h>
+
+LOG_MODULE_REGISTER(ifx_cat1_rtc, CONFIG_RTC_LOG_LEVEL);
+
+#define DT_DRV_COMPAT infineon_cat1_rtc
+
+#define _IFX_CAT1_RTC_STATE_UNINITIALIZED 0
+#define _IFX_CAT1_RTC_STATE_ENABLED       1
+#define _IFX_CAT1_RTC_STATE_TIME_SET      2
+
+#define _IFX_CAT1_RTC_INIT_CENTURY 2000
+#define _IFX_CAT1_RTC_TM_YEAR_BASE 1900
+
+#if defined(CONFIG_SOC_FAMILY_INFINEON_CAT1B)
+#if defined(SRSS_BACKUP_NUM_BREG3) && (SRSS_BACKUP_NUM_BREG3 > 0)
+#define _IFX_CAT1_RTC_BREG (BACKUP->BREG_SET3[SRSS_BACKUP_NUM_BREG3 - 1])
+#elif defined(SRSS_BACKUP_NUM_BREG2) && (SRSS_BACKUP_NUM_BREG2 > 0)
+#define _IFX_CAT1_RTC_BREG (BACKUP->BREG_SET2[SRSS_BACKUP_NUM_BREG2 - 1])
+#elif defined(SRSS_BACKUP_NUM_BREG1) && (SRSS_BACKUP_NUM_BREG1 > 0)
+#define _IFX_CAT1_RTC_BREG (BACKUP->BREG_SET1[SRSS_BACKUP_NUM_BREG1 - 1])
+#elif defined(SRSS_BACKUP_NUM_BREG0) && (SRSS_BACKUP_NUM_BREG0 > 0)
+#define _IFX_CAT1_RTC_BREG (BACKUP->BREG_SET0[SRSS_BACKUP_NUM_BREG0 - 1])
+#endif
+#endif
+
+#define _IFX_CAT1_RTC_BREG_CENTURY_Pos 0UL
+#define _IFX_CAT1_RTC_BREG_CENTURY_Msk 0x0000FFFFUL
+#define _IFX_CAT1_RTC_BREG_STATE_Pos   16UL
+#define _IFX_CAT1_RTC_BREG_STATE_Msk   0xFFFF0000UL
+
+static const uint32_t _IFX_CAT1_RTC_MAX_RETRY = 10;
+static const uint32_t _IFX_CAT1_RTC_RETRY_DELAY_MS = 1;
+
+static cy_stc_rtc_dst_t *_ifx_cat1_rtc_dst;
+
+#ifdef CONFIG_PM
+static cy_en_syspm_status_t _ifx_cat1_rtc_syspm_callback(cy_stc_syspm_callback_params_t *params,
+							 cy_en_syspm_callback_mode_t mode)
+{
+	return Cy_RTC_DeepSleepCallback(params, mode);
+}
+
+static cy_stc_syspm_callback_params_t _ifx_cat1_rtc_pm_cb_params = {NULL, NULL};
+static cy_stc_syspm_callback_t _ifx_cat1_rtc_pm_cb = {
+	.callback = &_ifx_cat1_rtc_syspm_callback,
+	.type = CY_SYSPM_DEEPSLEEP,
+	.callbackParams = &_ifx_cat1_rtc_pm_cb_params,
+};
+#endif /* CONFIG_PM */
+
+#define _IFX_CAT1_RTC_WAIT_ONE_MS() Cy_SysLib_Delay(_IFX_CAT1_RTC_RETRY_DELAY_MS);
+
+/* Internal macro to validate RTC year parameter falls within 21st century */
+#define IFX_CAT1_RTC_VALID_CENTURY(year) ((year) >= _IFX_CAT1_RTC_INIT_CENTURY)
+
+#define MAX_IFX_CAT1_CAL (60)
+
+/* Convert parts per billion to groupings of 128 ticks added or removed from one hour of clock
+ * cycles at 32768 Hz.
+ *
+ * ROUND_DOWN(ppb * 32768Hz * 60min * 60sec / 1000000000, 128) / 128
+ * ROUND_DOWN(ppb * 117964800 / 1000000000, 128) / 128
+ * ROUND_DOWN(ppb * 9216 / 78125, 128) / 128
+ */
+#define PPB_TO_WCO_PULSE_SETS(ppb) ((ROUND_DOWN((ppb * 9216 / 78125), 128)) / 128)
+
+/* Convert groupings of 128 ticks added or removed from one hour of clock cycles at
+ * 32768 Hz to parts per billion
+ *
+ * wps * 128 * 1000000000 / 32768Hz * 60min * 60sec
+ * wps * 128000000000 / 117964800
+ * wps * 78125 / 72
+ */
+#define WCO_PULSE_SETS_TO_PPB(wps) (wps * 78125 / 72)
+
+struct ifx_cat1_rtc_config {
+	uint32_t irqn;
+};
```

Comment:
this not not actually used, you can remove it, along with its instance in the CAT1_RTC_INIT macro :) save a whole 4 bytes of memory :D

Reply:
Great! Thank you @bjarki-andreasen. This final version of the "verification commit" should resolve this. As for calibration, the test has been ran and the results were as expected.